File: frontend/src/main/types/DashboardPortals/UsuarioPortalResponse.ts
// types/DashboardPortals/UsuarioPortalResponse.ts

export type RolPortal = 'ADMIN' | 'MIEMBRO';

export interface UsuarioPortalResponse {
  id: number;
  universidad: string;
  carrera: string;
  unidadAcademica: string | null;
  descripcion: string | null;
  logoUrl: string | null;
  iconoPortal: string | null;
  colorPortal: string | null;
  rol: RolPortal;
  cantidadMiembros: number;
  cantidadMaterias: number;
}

File: backend/src/main/kotlin/com/unsam/scholarium/dto/UsuarioPortalResponse.kt
package com.unsam.scholarium.dto

import com.unsam.scholarium.model.RolMembresia
import java.util.UUID

data class UsuarioPortalResponse(
    val id: Long,
    val universidad: String,
    val carrera: String,
    val descripcion: String?,
    val logoUrl: String?,
    val iconoPortal: String?,
    val colorPortal: String?,
    val rol: RolMembresia,
    val cantidadMiembros: Long,
    val cantidadMaterias: Long
)
